node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -265,57 +293,104 @@ impl ReceivableDao for ReceivableDaoReal {
+                 where wallet_address = :wallet";
+        let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, \
+                 balance_low_b = :balance_low_b, last_received_timestamp = :last_received \
+                 where wallet_address = :wallet";
+
+        match received_payments.iter().try_for_each(|received_payment| {
+            let last_received_timestamp = to_time_t(timestamp);
+            let params = SQLParamsBuilder::default()
+                .key(WalletAddress(&received_payment.from))
+                .wei_change(Subtraction("balance", received_payment.wei_amount))
+                .other_params(vec![Param::BothClauses((
+                    ":last_received",
+                    &last_received_timestamp,
+                ))])
+                .build();
+
+            let result = big_int_db_processor.execute(
+                Either::Right(txn.as_ref()),
+                BigIntSqlConfig::new(main_sql, overflow_update_clause, params),
+            );
+
+            match result {
+                Ok(_) => Ok(()),
+                Err(BigIntDatabaseError::General(err_msg)) => {
+                    Err(ReceivableDaoError::RusqliteError(err_msg))
+                }
+                Err(BigIntDatabaseError::RowChangeMismatch { .. }) => {
+                    Self::verify_possibly_unknown_wallet(
+                        txn.as_ref(),
+                        logger,
+                        received_payment,
+                        received_payments,
+                    )
+                }
             }
+        }) {
+            Ok(_) => Ok(txn),
+            Err(e) => Err(e),
         }
-        match xactn.commit() {
-            // Error response is untested here, because without a mockable Transaction, it's untestable.
-            Err(e) => Err(ReceivableDaoError::RusqliteError(format!("{:?}", e))),
-            Ok(_) => Ok(()),
+    }
+
+    fn verify_possibly_unknown_wallet(
+        txn: &dyn TransactionWrapper,
+        logger: &Logger,
+        suspect: &BlockchainTransaction,
+        all_received_payments: &[BlockchainTransaction],
+    ) -> Result<(), ReceivableDaoError> {
+        let amount = suspect.wei_amount;
+        let address = &suspect.from;
+        let block_number = suspect.block_number;
+
+        if Self::check_row_presence(txn, address) {
+            Self::more_money_received_roll_back_error_log(logger, all_received_payments);
+            Err(ReceivableDaoError::RusqliteError(format!(
+                "Update for received payment with {amount} wei ran without producing a data change, \
+                despite a record for the wallet {address} exists"
+            )))
+        } else {
+            info!(
+                logger,
+                "Detected an inward transfer of {amount} wei in the block {block_number}. Address \
+                {address} does not match any of the tracked debtors. Ignoring"
+            );
```

Comment:
Maybe rephrase this which is closely related to the name of the card, something like:


```
Received {amount} wei from unknown debtor {address} in the block {block_number}.
```